client/app/scripts/superdesk-authoring/authoring.js

```diff
@@ -1382,11 +1397,7 @@
                 };
 
                 scope.$watch('options', function() {
```

Comment:
it could be just `scope.$watch('options', fetchTemplates, true);` - no need to wrap it into extra function